### DISCUSSION TITLE
This Serverless Function has crashed. 500: INTERNAL_SERVER_ERROR

### DISCUSSION THREAD

**Discussion by farhannilok** [vercel/community#5231] [Help]
### Summary

I deployed express api on vercel using vercel CLI with some mongoDB code it says the serverless function crushed. But the app runs locally very well then i checked the logs and it says cannot find module 'express'. I revalidate all the packages and it seems fine.
The log error is this.
Cannot find module 'express'
Require stack:
- /var/task/index.js
- /var/task/___now_launcher.js
- /var/runtime/index.mjs
Did you forget to add it to "dependencies" in `package.json`?
INIT_REPORT Init Duration: 190.64 ms	Phase: invoke	Status: error	Error Type: Runtime.ExitError
Error: Runtime exited with error: exit status 1
Runtime.ExitError

### Steps to Reproduce

![image](https://github.com/vercel/community/assets/85245133/2f592214-5a24-4468-b1c5-9317f828afdc)

package.json file:

{
  "name": "car-doctor-server",
  "version": "1.0.0",
  "description": "",
  "main": "index.js",
  "scripts": {
    "start": "node index.js",
    "test": "echo \"Error: no test specified\" && exit 1"
  },
  "keywords": [],
  "author": "",
  "license": "ISC",
  "dependencies": {
    "cors": "^2.8.5",
    "dotenv": "^16.3.1",
    "express": "^4.18.2",
    "jsonwebtoken": "^9.0.2",
    "mongodb": "^6.3.0"
  }
}

**Comment by amyegan** [comment]
Hey @Farhan-Hasan-Nilok! We have a guide that covers [basic Express app setup on Vercel](https://vercel.com/guides/using-express-with-vercel#standalone-express) and some [drawbacks and edge cases](https://vercel.com/guides/using-express-with-vercel#drawbacks-and-edge-cases) to consider. You can find that guide at [vercel.com/guides/using-express-with-vercel](https://vercel.com/guides/using-express-with-vercel).

I built my own Express.js example using that guide. You can find that repo at [github.com/amyegan/express-vercel](https://github.com/amyegan/express-vercel)
